### PR TITLE
refactor: enforce tflint naming convention `snake_case`

### PR DESCRIPTION
**Pull request by jabielecki**
## Description

Enforce tflint naming convention `snake_case` instead of `dashed-names` or `UPPERCASE`.

Enforce pinning down the external module sources, instead of taking those from the `master` branch.

## Motivation and Context

Ease the code review process - the devs will be notified by CI job about basic code issues, not by a human.

## How Has This Been Tested?

Example `init` and `apply` per README.

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

File: examples/vmseries_scaleset/README.md [hunk 1]
```diff
@@ -8,9 +8,19 @@ This folder shows an example of Terraform code that helps to deploy an auto-scal
 
 Create a `terraform.tfvars` file and copy the content of `example.tfvars` into it, adjust if needed.
 
-```bash
-$ terraform init
-$ terraform apply
+Create a `files/init-cfg.txt` file and copy the content of `files/init-cfg.sample.txt` into it, adjusting it. (Do not add the file to the repository if it contains secrets such as vm-auth-key.)
```
Comment (jabielecki) [inline examples/vmseries_scaleset/README.md:11]: The `.sample` files are named so to signify they are for education.

Do not try to use them directly, and do not imply that users should
modify them for their further continued use. Tell users how to proceed.

File: modules/vmss/outputs.tf [hunk 2]
```diff
@@ -1,3 +1,4 @@
+# tflint-ignore: terraform_naming_convention # TODO rename to scale_set_name, but bundle with next breaking change
```
Comment (jabielecki) [inline modules/vmss/outputs.tf:1]: I simply don't want to pollute the user-facing changelog with a lot of breaking changes to `vmss`. It's far from being usable.

**Review by jabielecki — COMMENTED** [review]
Self-comments.

**Review by damianfedeczko — APPROVED** [review]
LGTM 👍

**Pull request merged**